contracts/MessagePort.sol

```diff
@@ -101,34 +106,42 @@ contract MessagePort is IMessagePort, Ownable2Step {
         uint64 fromChainId_,
         address fromDappAddress_,
         address toDappAddress_,
-        bytes memory messagePayload_
+        bytes memory messagePayloadWithId_
     ) external {
         require(
             localDockExists(fromChainId_, msg.sender),
             "Local dock not exists"
         );
 
+        (uint256 messageId, bytes memory messagePayload_) = abi.decode(
+            messagePayloadWithId_,
+            (uint256, bytes)
+        );
+
         try
             IMessageReceiver(toDappAddress_).recv(
                 fromChainId_,
                 fromDappAddress_,
-                messagePayload_
+                messagePayload_,
+                messageId
```

Comment:
messageId is used by message indexer, do not expose to dAPP user.

Reply:
Does dApp not need messageId? After sending a message, dApp can use this ID to check the status of the message in the explorer

Reply:
Explorer is off-chain service,  dAPP is on-chain, they have no relationship.